### PR TITLE
Fix #119

### LINKED ISSUE #119: auto-enter VR causes instructions to dissapear
After entering VR and refreshing the page, the instructions disappear.

This bit of code is responsible for this:
https://github.com/aframevr/a-saturday-night/blob/master/src/ui2d.js#L19-L21

This issue presents itself when trying to upgrade to a-frame 6.1.  See PR https://github.com/aframevr/a-saturday-night/pull/118 and https://github.com/aframevr/a-saturday-night/pull/117

### PR DESCRIPTION
**Pull request by fernandojsg**
/cc @caseyyee

### CODE REVIEW

**Comment by caseyyee** [comment]
@fernandojsg if you're playing back a recording,  you have to  _start_ to set game state, so we can't just hide it.

**Comment by cvan** [comment]
friendly bump, @fernandojsg

**Comment by fernandojsg** [comment]
@cvan @caseyyee @dmarcos I'm still not sure how we want a-saturday to behave when loading a dance:

1. We could just show the main screen and wait for the user to click Start and it will enter vr and start the playback. This is how is working right now, we should just fix the issue when auto-entering VR because of link traversal.

2. We could skip that main screen and just start the playback right away. What I like about this option is that we don't need user interaction and it's suitable for experiences with link traversal: you're on a site click a link and watch the dance directly.

My vote is for the 2nd, what do you think?

**Comment by fernandojsg** [comment]
Ok, I've go ahead with the 2nd option:

Here you can see that I load a dance url but as I was not on VR it shows the instructions screen, I click Enter, click enter VR and I refresh the page, so I'll reload the same url but as I was before on VR it just enter directly on VR without showing the instructions and start playing the dance.

File: src/components/states/instructions.js [hunk 1]
```diff
@@ -6,22 +6,40 @@ AFRAME.registerComponent('instructions', {
...
+
+    if (inVR) {
+      this.onStartButtonClicked();
+    } else {
+      var self = this;
+      window.addEventListener('vrdisplayactivate', function () {
+        if (sceneEl.getAttribute('game-state').state === 'instructions') {
+          self.onStartButtonClicked();
+        }
+        //@todo remove eventlistener?
```
Comment (dmarcos) [inline src/components/states/instructions.js]: Is not the button removed start replaying?

File: src/components/states/instructions.js [hunk 2]
```diff
@@ -6,22 +6,40 @@ AFRAME.registerComponent('instructions', {
...
+    var sceneEl = this.el.sceneEl;
+    var inVR = sceneEl.is('vr-mode');
+
+    if (inVR) {
+      this.onStartButtonClicked();
+    } else {
+      var self = this;
+      window.addEventListener('vrdisplayactivate', function () {
+        if (sceneEl.getAttribute('game-state').state === 'instructions') {
+          self.onStartButtonClicked();
```
Comment (dmarcos) [inline src/components/states/instructions.js]: I would call the method `replayDance` or `startReplaying`

Comment (fernandojsg) [inline src/components/states/instructions.js]: Done

File: src/components/states/instructions.js (hunk 1 above)
Comment (fernandojsg) [inline src/components/states/instructions.js]: We hide the whole UI when playing but what I mean there is if we should remove the `vrdisplayactivate` listener too.
Probably it doesn't matter as once we're presenting we can't go back to the instructions so the case of that listener being triggered several times won't happen either

File: src/components/states/instructions.js [hunk 3]
```diff
@@ -6,22 +6,44 @@ AFRAME.registerComponent('instructions', {
...
+    var inVR = sceneEl.is('vr-mode');
+
+    if (!sceneEl.hasLoaded) {
+      sceneEl.addEventListener('loaded', this.handleDanceData.bind(this));
+      return;
+    }
+
+    if (inVR) {
+      this.startReplaying();
+    } else {
```
Comment (dmarcos) [inline src/components/states/instructions.js]: I would attach a listener to `enter-vr` on the init method that starts replaying if any data has loaded.

Comment (fernandojsg) [inline src/components/states/instructions.js]: But what if the `enter-vr` event fires before the data is loaded?

**Comment by dmarcos** [comment]
Thanks!

**Pull request merged**